### ISSUE TITLE
Tracking issue for version 0.1

### ISSUE THREAD

**Issue by RDambrosio016** [RDambrosio016/RSLint#4]
Just a brief list of things id like to add before 0.1:
Rules:
- [x] no-extra-boolean-cast - (#15) - finished
- [x] no-prototype-builtins - [finished](https://github.com/RDambrosio016/RSLint/commit/76093da22739549b38a72a950ca383da48cf9f5c)
- [x] no-setter-return (#9) - finished
- [x] no-sparse-arrays - [finished](https://github.com/RDambrosio016/RSLint/commit/1ee5e39eb56fc8a7a2552f99e4497fc64b68cdfd)
- [x] no-unexpected-multiline [finished](https://github.com/RDambrosio016/RSLint/commit/cc21597f7015f1096c75ae13adcf78a9afd21100)
- [x] use-isnan (#13) - finished
- [x] valid-typeof (#8) - finished

These are most of the eslint-reccomended rules for errors which dont rely on regex parsing, scope analysis, and code flow analysis, id like all those tools to be developed as individual analyzers instead of rushing them for rules. 

Fixes:
- [x] `rslint_parser::numbers::parse_js_num` needs to account for legacy octal literals and needs tests - [finished](https://github.com/RDambrosio016/RSLint/commit/6e54c8208ad53d172dcb5095646c82040b7d8976)
- [x] no-inner-declaration wrongfully fails (#12) - finished

Misc:
- [x] Add a changelog to keep track of features and fixed bugs (starting from 0.1) - finished

@magurotuna if you'd like to do more in depth stuff with rules then working on any of these rules or fixes is a great start! 😄 
although if you start work on a rule please put a comment saying which rule just so we don't accidentally work on the same things.

**Comment by magurotuna** [comment]
@RDambrosio016 Great instruction! sounds good place to start. Yes, I'd love to do more things 😃 Then I'll work on `no-setter-return` rule first.

**Comment by RDambrosio016** [comment]
Great! `getter-return` has a lot of common logic so i would check it out, particularly `.structural_lossy_token_eq(&["Object", ".", "defineProperty"])` might be of use.

**Comment by RDambrosio016** [comment]
@magurotuna i changed the AST def for Expr a little bit in the most recent commit (NAME => NAME_REF for expr), you might want to pull so you don't use `Expr::Name` which would be incorrect.

**Comment by magurotuna** [comment]
@RDambrosio016 
> i changed the AST def for Expr a little bit in the most recent commit (NAME => NAME_REF for expr)

Thanks for letting me know that! I'll pull the latest commit before starting to work.

**Comment by Stupremee** [comment]
I would like to try to implement the ``no-extra-boolean-cast` lint

**Comment by RDambrosio016** [comment]
Go ahead! i had some code from when i tried implementing it before scrapping it because it did not easily work for things like grouping expressions, the rule needs a top-down approach, not bottom-up, you can however take the tests from it and use them 🙂 https://paste.rs/3o2.rs

**Comment by RDambrosio016** [comment]
I think #12 is serious enough that it should be fixed before 0.1, therefore i have added it to the list, not sure how it slipped through 🤔, moreover it uses nearly the same logic as eslint.

**Comment by RDambrosio016** [comment]
The rules and fixes have all been implemented, i just need to change some random things like making xtask non-published and always v0.0.0 and add changelogs to each crate

**Comment by RDambrosio016** [comment]
Version 0.1 has been released! https://github.com/RDambrosio016/RSLint/commit/eb4dee150fb6018cf5a97b3e8844a38c2c75339c

the dev branch will still be used for development and will be merged into master at release. From now on i would like to document any changes in the appropriate changelog file in each major crate. I will also be publishing most of the crates to crates.io now.
